controllers/admin/AdminTranslationsController.php

```diff
@@ -1581,13 +1584,16 @@ public function postProcess()
      *
      * @param bool $save_and_stay : true if the user has clicked on the button "save and stay"
      * @param bool $conf : id of confirmation message
+     * @param bool $modify_translation : true if the user has clicked on the button "Modify translation"
      */
-    protected function redirect($save_and_stay = false, $conf = false)
+    protected function redirect($save_and_stay = false, $conf = false, $modify_translation = false)
     {
         $conf = !$conf ? 4 : $conf;
         $url_base = self::$currentIndex.'&token='.$this->token.'&conf='.$conf;
-        if ($save_and_stay) {
-            Tools::redirectAdmin($url_base.'&lang='.$this->lang_selected->iso_code.'&type='.$this->type_selected.'&theme='.$this->theme_selected);
+        if($modify_translation){
+            Tools::redirectAdmin(self::$currentIndex.'&token='.$this->token.'&lang='.$_REQUEST['langue'].'&type='.$this->type_selected.'&module='.$_REQUEST['module'].'&theme='.$this->theme_selected);
```

Comment:
langue? $_REQUEST? i think something is wrong here and i'm not sure if it's a good place for this, i think that you can check if user modifies translation of the module before this method